src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -337,23 +336,31 @@ private static final class RuntimeFields {
         private static final long queueSizeOffset;
 
         static {
-            int cacheLine = CacheInfo.getSmallestDataCacheLineSize();
-            if (cacheLine == 0) {
-                // guess
-                cacheLine = 64;
-            }
-            // cpu spatial prefetcher can drag 2 cache-lines at once into L2
-            int pad = cacheLine > 128 ? cacheLine : 128;
             int longScale = unsafe.arrayIndexScale(long[].class);
             int taskNodeScale = unsafe.arrayIndexScale(TaskNode[].class);
-            // these fields are in units of array scale
-            unsharedTaskNodesSize = pad / taskNodeScale * (numUnsharedObjects + 1);
-            unsharedLongsSize = pad / longScale * (numUnsharedLongs + 1);
-            // these fields are in bytes
-            headOffset = unsafe.arrayBaseOffset(TaskNode[].class) + pad;
-            tailOffset = unsafe.arrayBaseOffset(TaskNode[].class) + pad * 2;
-            threadStatusOffset = unsafe.arrayBaseOffset(long[].class) + pad;
-            queueSizeOffset = unsafe.arrayBaseOffset(long[].class) + pad * 2;
+            if (ProcessorInfo.availableProcessors() > 1) {
+                // this is fine for pretty much 32 and 64 bit x86 and ARM processors; see
+                // https://github.com/ziglang/zig/blob/0.13.0/lib/std/atomic.zig#L424-L434
+                // cpu spatial prefetcher can drag 2 cache-lines at once into L2
+                int pad = 128;
+                // we both pad before and after the array to avoid false sharing with surrounding heap objects
+                unsharedTaskNodesSize = pad / taskNodeScale * (numUnsharedObjects + 1);
+                unsharedLongsSize = pad / longScale * (numUnsharedLongs + 1);
+                // these fields are in bytes
+                headOffset = unsafe.arrayBaseOffset(TaskNode[].class) + pad;
+                tailOffset = unsafe.arrayBaseOffset(TaskNode[].class) + pad * 2;
+                threadStatusOffset = unsafe.arrayBaseOffset(long[].class) + pad;
+                queueSizeOffset = unsafe.arrayBaseOffset(long[].class) + pad * 2;
+            } else {
+                unsharedTaskNodesSize = numUnsharedObjects;
```

Comment:
@dmlloyd we have no coverage for this unless we run our test suite with `-XX:ActiveProcessorCount=1`
so please take a look if I did something wrong